### PR TITLE
util/log: remove rotation on SIGUSR2; flush on SIGHUP instead of SIGUSR1

### PR DESCRIPTION
**Pull request by benesch**
This partially reverts commit 2b91e0f55f2eaa7cbb0c24dcc2f931923007a36b
per the discussion in [0]:

  * Removes log rotation upon receiving SIGUSR2. Ben pointed out that
    external processes wishing to trigger a log rotation will also need
    a notification that the rotation has completed. SIGUSR2 is delivered
    asychronously, making this difficult.

  * Triggers log flushing on SIGHUP instead of SIGUSR1. We already use
    SIGHUP to trigger a certificate reload, so we might as well let
    SIGHUP trigger all non-disruptive changes to leave SIGUSR1/SIGUSR2
    open for future use.

[0]: https://github.com/cockroachdb/cockroach/pull/23801#issuecomment-372792834.

Release note (general change): a CockroachDB process will now flush its
logs upon receiving `SIGHUP`.

Note to docs team: The previous release note about flushing and rotating
logs in response to SIGUSR1 and SIGUSR2, respectively, is null and void.

### CODE REVIEW

**Comment by bdarnell** [comment]
The subsequent commit 4ff0363cafd181bf967330320d2174897e401011 depends on this. I think if we introduced flushing on SIGHUP we could use that from the test.

**Comment by bdarnell** [comment]
:lgtm:, but wait for @knz to weigh in too. 

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23803#-:-L7W3fcQExwBvOvbfbGr:b4r8irt)*

**Comment by benesch** [comment]
Required a little bit more finagling to get tests to pass, but nothing major. @knz, awaiting your review!

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23803#-:-L7W64TU1H69H9xJMaPq:b-cemyja)*

File: pkg/cli/interactive_tests/test_flags.tcl [hunk 1]
```diff
@@ -1,32 +0,0 @@
-#! /usr/bin/env expect -f
```
Comment (knz) [inline pkg/cli/interactive_tests/test_flags.tcl]: This file should not have been removed.

File: pkg/cli/interactive_tests/test_rotate_flush.tcl [hunk 2]
```diff
@@ -1,24 +0,0 @@
-#! /usr/bin/env expect -f
```
Comment (knz) [inline pkg/cli/interactive_tests/test_rotate_flush.tcl]: This file should remain and test the flush.

**Comment by knz** [comment]
Please rephrase the title of the PR
also add a release note  that corrects the previous one, so that the person who collects the release notes know what to do.

**Comment by knz** [comment]
Out of curiosity, can multiple goroutines listen to the same signal with signal.Notify? Is it guaranteed to deliver the same received signal to all of them?

File: pkg/cli/interactive_tests/test_rotate_flush.tcl (hunk 2 above)
Comment (benesch) [reply pkg/cli/interactive_tests/test_rotate_flush.tcl]: I adjusted `flush_server_logs` to wait for the flush, so every caller of `flush_server_logs` implicitly tests the function. Seemed superfluous to have both given how expensive each of these tests is, but can add it back if you feel strongly!

**Review by benesch — COMMENTED** [review]
> Out of curiosity, can multiple goroutines listen to the same signal with signal.Notify? Is it guaranteed to deliver the same received signal to all of them?

Yep! From the godoc:

> It is allowed to call Notify multiple times with different channels and the same signals: each channel receives copies of incoming signals independently.

File: pkg/cli/interactive_tests/test_rotate_flush.tcl (hunk 2 above)
Comment (knz) [reply pkg/cli/interactive_tests/test_rotate_flush.tcl]: No it's ok. thanks

**Comment by benesch** [comment]
Ok, done. TFTR!

**Comment by knz** [comment]
:+1: :shipit:

**Pull request merged**